internal/services/riotclient.go

```diff
@@ -0,0 +1,57 @@
+package services
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"riot-developer-proxy/internal/valueobjects"
+)
+
+type RiotClient struct {
+	client    http.Client
+	baseURI   string
+	riotToken *string
+}
+
+func NewRiotClient(client http.Client, baseURI string) *RiotClient {
+	return &RiotClient{
+		client:  client,
+		baseURI: baseURI,
+	}
+}
+
+func (rc *RiotClient) WithToken(token string) {
```

Comment:
We generally like this pattern of writing wrapper setter functions.

Reply:
I like this pattern too! It's based at builder pattern, but for simplified contexts!